modules/data-entry/src/main/java/io/uhndata/cards/internal/RequiredSubjectTypesValidator.java

```diff
@@ -0,0 +1,138 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.uhndata.cards.internal;
+
+import java.util.*;
+
+import org.apache.jackrabbit.oak.api.CommitFailedException;
+import org.apache.jackrabbit.oak.api.Type;
+import org.apache.jackrabbit.oak.spi.commit.DefaultValidator;
+import org.apache.jackrabbit.oak.spi.commit.Validator;
+import org.apache.jackrabbit.oak.spi.state.NodeState;
+import org.apache.sling.api.resource.LoginException;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.resource.ResourceResolverFactory;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RequiredSubjectTypesValidator extends DefaultValidator
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(RequiredSubjectTypesValidator.class);
+
+    private final ResourceResolverFactory rrf;
+
+    public RequiredSubjectTypesValidator(ResourceResolverFactory rrf)
+    {
+        this.rrf = rrf;
+    }
+
+    @Override
+    public @Nullable Validator childNodeAdded(String s, NodeState after) throws CommitFailedException
+    {
+        // Get the type of this node. Return immediately if it's not a cards:Form node
+        final String childNodeType = after.getName("jcr:primaryType");
+        if (!("cards:Form".equals(childNodeType))) {
+            return this;
+        }
+
+        // Get the jcr:uuid values for the Form's associated Questionnaire and Subject
+        final String questionnaireUUID = after.getProperty("questionnaire").getValue(Type.REFERENCE).toString();
+        final String subjectUUID = after.getProperty("subject").getValue(Type.REFERENCE).toString();
+        LOGGER.warn("Added this --> {}", after);
+        LOGGER.warn("A cards:Form node was just added with questionnaire={} and subject={} !!!", questionnaireUUID,
+                subjectUUID);
+        // Obtain a ResourceResolver for querying the JCR
+        final Map<String, Object> parameters =
+            Collections.singletonMap(ResourceResolverFactory.SUBSERVICE, "requiredSubjectTypesValidator");
+        try (ResourceResolver serviceResolver = this.rrf.getServiceResourceResolver(parameters)) {
+
+            // Get the Subject Resource associated with this Form
+            final Resource subject = getSubjectResourceByUuid(serviceResolver, subjectUUID);
+            LOGGER.warn("Resolved /Subjects/{} --> {}", subjectUUID, subject);
+            if (subject == null) {
+                return this;
+            }
+
+            // Get the type value for the Subject
+            final String type = subject.getValueMap().get("type", "null");
```

Comment:
```suggestion
            final String type = subject.getValueMap().get("type", "");
```
would be more consistent with how this is done in other places in the project.